### ISSUE TITLE
Locking a class's scheduling assignment

### ISSUE THREAD

**Issue by jmoldow** [learning-unlimited/ESP-Website#460]
From MIT: "the ability to lock a class in Ajax.  Obviously this is something any admin could undo, but they would see the name of the person who locked it.  This would help reduce mistakes with, for example, weirdly constrained classes."

Perhaps the lock could come with a comment field, so you can leave a note for others about why the class is locked.

**Comment by uakfdotb** [comment]
I implemented something like this separately from a palette feature. Basically there's a button "Set Comment" in the top right of AJAX scheduler; it opens a dialog where you can write a scheduling-related comment and optionally lock the current scheduling assignment. The comment/lock are specific to the AJAX scheduler, and the lock is only checked client-side. To unlock you hit the same button and uncheck the box in the dialog.
